### ISSUE TITLE
Include nested of nested association

### ISSUE THREAD

**Issue by blocknotes** [Netflix/fast_jsonapi#47]
It would be great to allow to include nested of nested associations... example:

```
class Post < ApplicationRecord
  has_many :post_tags, dependent: :destroy
  has_many :tags, through: :post_tags
  belongs_to :author, touch: true
end

class Author < ApplicationRecord
  has_many :posts, dependent: :nullify
  has_one :profile, dependent: :destroy
end

class Profile < ApplicationRecord
  belongs_to :author, touch: true
end
```

In Post serializer, to include author's profile something like:
`PostSerializer.new @post, {include: [:author, :'author.profile', :tags]}`
could be very useful.

**Comment by potato20** [comment]
+1 wud rly lik dis vry much pls netflox aprcite it

**Comment by christophersansone** [comment]
Me too!  IIRC, AMS would specify it as `{ include: [ { author: [ :profile ] }, :tags ] }`.  This approach seems ideal because the developer can define a deeply nested hash and arrays to reflect the deeply nested relationships.

**Comment by shishirmk** [comment]
@christophersansone Agree. Using the AMS approach would be ideal. We would like to have this feature added to fast_jsonapi as soon as we can. @potato20 @blocknotes pull requests are welcome

**Comment by SumLare** [comment]
Trying to make it work right now, it's kinda related to #67. Should I wait for it to be merged first?

**Comment by shishirmk** [comment]
@SumLare ya i think there are a lot of changes coming up so do you mind waiting a little bit. We are overwhelmed by the response and would love to add all the features that are part of the JSON:API spec. 

Once some of the relevant pull requests are merge to dev go ahead and create one with all your changes.

**Comment by SumLare** [comment]
@shishirmk yea sure looking forward to help you guys.

**Comment by NickBeukema** [comment]
@SumLare is there anything you need assistance with in completing this? Currently working on a project that would benefit from this, and would love to help if needed.

**Comment by SumLare** [comment]
@NickBeukema I’ll try to make it work by the end of this weekend, didn’t have time lately

**Comment by jshow** [comment]
I'd love to help as well - this is the main blocker to replacing AMS and basking in the velocity of fast_jsonapi

**Comment by shishirmk** [comment]
@jshow @SumLare @NickBeukema i think this would be a good time to attack this problem. Have any of you started working on it?

**Comment by jshow** [comment]
@shailesh33 , I've put together a PR for discussion

https://github.com/Netflix/fast_jsonapi/pull/152

**Comment by shishirmk** [comment]
This should be released in version 1.2 of the gem
